src/ui/components/Events/Event.tsx

```diff
@@ -31,15 +162,142 @@ export const getEventLabel = (event: ReplayEvent) => {
   return label;
 };
 
+const FIND_MOUSE_EVENT_TARGET_AND_REACT_CLICK_HANDLER = `
+  // Outer body runs in scope of the "current" click handler.
+  // Grab the click handler's arguments.
+  const args = [...arguments];
+  
+  (function findEventTargetAndClickHandler() {
+    // One of the args should be a mouse event
+    const foundMouseEvent  = args.find(a => typeof a === "object" && a instanceof MouseEvent);
+
+    if (foundMouseEvent) {
+      // We _could_ probably just return the onClick function or undefined here
+      const res = {
+        target: event.target,
+      }
+
+      // Search the event target node and all of its ancestors
+      // for React internal props data, and specifically look
+      // for the nearest node with an onClick prop if any.
+      let currentNode = event.target;
+      while (currentNode) {
+        const keys = Object.keys(currentNode)
+        const reactPropsKey = keys.find(key => {
+          return (
+            key.startsWith("${REACT_16_EVENT_LISTENER_PROP_KEY}") || 
+            key.startsWith("${REACT_17_18_EVENT_LISTENER_PROP_KEY}")
+          )
+        });
+
+        if (reactPropsKey) {
+          const props = currentNode[reactPropsKey];
+          if (props.onClick) {
+            res.onClick = props.onClick;
+            res.handlerNode = currentNode;
```

Comment:
This matches props that are literally named `onClick` ?

I think this means it will always find the one attached to the DOM element? What location would it return for the following scenario?
```jsx
function SomeGenericComponent({ handleClick }) {
  return (
    <div onClick={handleClick}>
      {/* ... */}
    </div>
  );
}
```

What about a scenario like this?
```jsx
function noop() {}

<div onClick={outerHandler}>
  <div onClick={enabled ? innerHandler : noop}>
    <button>Click target</button>
  </div>
</div>
```

I think we'd find `noop` and not `outerHandler`? (The above example is stupid but I think this pattern sometimes exists in actual apps– just further apart in the React tree.)

Reply:
Yes, literally `onClick`, because that's what the `props` object ends up as: `{onClick: someFunctionHere}`.

Reply:
Note to self:
What happens if there's an imperative handler that uses capture higher up in the DOM? and what if it stops prop? Will we incorrectly tag a nested (declarative) handler in that case?